qutebrowser/misc/throttle.py

```diff
@@ -0,0 +1,92 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2018 Jay Kamat <jaygkamat@gmail.com>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""A simple throttling decorator."""
+
+import typing
+import time
+import functools
+
+from qutebrowser.utils import usertypes
+
+
+class throttle:  # noqa: N801,N806 pylint: disable=invalid-name
+
+    """A simple function decorator to throttle calls.
+
+    If a request comes in, it will be processed immediately. If another request
+    comes in too soon, it is ignored, but will be processed when a timeout
+    ends. If another request comes in, it will update the pending request.
+
+    """
+
+    def __init__(self, throttle_ms: int) -> None:
```

Comment:
There's a lot of different usages of the word "throttle" here - why not `delay_ms`?